### PR TITLE
typing ir.py - part 5

### PR DESCRIPTION
**Pull request by aorenste**
See #131852 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131852
* #131851
* #131850
* __->__ #131849
* #131848
* #131847
* #131846
* #131845

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/ir.py [hunk 1]
```diff
@@ -4363,39 +4388,40 @@ def create(cls, inputs, dim):
...
+    def can_realize_into_without_copy(cls, src: IRNode) -> bool:
         if isinstance(src, TensorBox):
             # unwrap a TensorBox
             return cls.can_realize_into_without_copy(src.data)
 
-        return isinstance(src.data.layout, FlexibleLayout) and not isinstance(
-            src.data, ExternKernelAlloc
+        return isinstance(src.data.layout, FlexibleLayout) and not isinstance(  # type: ignore[attr-defined]
+            src.data, ExternKernelAlloc  # type: ignore[attr-defined]
         )
```
Comment (eellison) [inline torch/_inductor/ir.py:4421]: Feel like we should be able to type this but it's a bit annoying

Comment (eellison) [reply torch/_inductor/ir.py:4421]: It's only ever invoked with StorageBox as far as I can tell (and as far as runtime collected types say too: https://github.com/eellison/pyannotate/blob/master/inductor_collected_types.json)

**Review by eellison — APPROVED** [review]

**Comment by Skylion007** [comment]
@aorenste Did we ever get this rest of this typing stack merged? Saw that there was a transient merge error during the initial merge

**Comment by aorenste** [comment]
> @aorenste Did we ever get this rest of this typing stack merged? Saw that there was a transient merge error during the initial merge

I didn't. Every once in a while I poke at these but I've mostly been working on other things.

**Pull request closed without merging**